### ISSUE TITLE
[BUG] Phone freezes for good few minutes

### ISSUE THREAD

**Issue by Lincoln44** [ananjaser1211/hadesRom_Reports#17]
[Log_2020-12-29_23-46-26.txt](https://github.com/ananjaser1211/hadesRom_Reports/files/5752614/Log_2020-12-29_23-46-26.txt)
**Describe the bug**
When trying to open some apps (doesn't matter which ones, because it happens on all of them, not just only with poweramp as I suspected before) the phones starts to lagg at first. Then it freezes and its like dead. I can't do nothing for good few minutes. The screen becomes unresponsive. Power, volume, other buttons don't react either. After a minute or so, the phones becomes responsive again.

P.s. I had experienced this since I installed this rom 2 days ago. However, for the first time today, I got system ui crash after this freeze I explained above. Hopefully, you can see that in the provided logs.

**Info about the rom instalation and mods:**

I didn't install any bloatware. No Bixby, no dual speaker mod, no other bs. No root either, so no mods whatsoever. Clean and simple instalation.

**Smartphone (please complete the following information):**
 - Device: Galaxy S8+
 - Model: G955F

**Comment by SoRadGaming** [comment]
Exact same issue, sometimes I take a messenger call and the whole phone freezes, or I just open youtube and system freeze. Ill try to grab a log next time it happens.

**Comment by nurikan** [comment]
My intuition says this is related to audio problem or vice versa.

**Comment by ananjaser1211** [comment]
Do you have a solid way to re-create this sysyemUI crash / freeze ?

**Comment by corsicanu** [comment]
@Lincoln44 Last time i seen that error from your log it was caused by xposed/riru and some other incompatible addons. 
Humor me and reflash the rom without wipes, this time don't install Magisk, and try to reproduce the issue. Thanks

LE: Just seen you said you're not rooted. Can you try disabling GoodLock modules if you have installed?

**Comment by ananjaser1211** [comment]
Any themes installed ?

**Comment by nurikan** [comment]
> Any themes installed ?

No themes, no bloatware 
Clean install
Periodic hard freezes sometimes even ui crash

**Comment by ananjaser1211** [comment]
Since it is periodic, could you use logging kit to keep the phone recording, so when it crashes it gets captured.

in OPs log, his phone is trying to load an icon on the launcher, that is not really an icon. and it crashes TW Home & the systemUI

```
12-29 23:46:56.972 29031 30363 E IconLoader: Could not find icon drawable from resource

12-29 23:46:56.972 29031 30363 E IconLoader: android.content.res.Resources$NotFoundException: Resource ID #0xffffffff

12-29 23:46:56.972 29031 30363 E IconLoader: 	at android.content.res.ResourcesImpl.getValueForDensity(ResourcesImpl.java:285)
```

Since none of these apps are modified in any way, and this error only seems to happen to few (3 so far, 1 on telegram who had themes), i have to assume it is some app that gets installed, or theme that causes this.

so we will need alot more info, including an active log to capture the crash

**Comment by corsicanu** [comment]
Maybe would worth mentioning Poweramp version and type - free/pro/cracked

**Comment by nurikan** [comment]
> Maybe would worth mentioning Poweramp version and type - free/pro/cracked

I dont have any poweramp but have same issue
But sometimes it occurs during wathcing videos (video player related something)
Anyway i will try to catch the logs))

**Comment by Lincoln44** [comment]
Poweramp is legit for me. Version - build 893 arm64. 

I don't use cracked apps, except for modified Spotify and plex. 


Perhaps this is some key info, not sure. I had installed Goodlock from galaxy store, not through rom instalation. However, I had this issue without and with goodlock installed. Tbh, a few modules are installed of goodlock, but I don't use them at all. So I don't think that's the issue. 

About root. Yes, I didn't have root at all, but I had to root to be able to provide the logs according to the tutorial, because my cable is not original and PC wont recognize my phone for ADB commands. 

Quick point to note: before I had root, I had this periodic freeze, but never system ui crash. After I rooted yesterday and then tried to catch the freeze, I got system ui crash message just before the phone unfroze after a minute or so for the very first time.

I think I mentioned everything now. Hopefully all this helps.

Should I unroot and reflash the rom like corsi said?

**Comment by ananjaser1211** [comment]
no freezes on legit poweramp here, i also have cracked plex.

cant recreate any of the freezes and crashes yet

**Comment by Lincoln44** [comment]
> no freezes on legit poweramp here, i also have cracked plex.
> 
> cant recreate any of the freezes and crashes yet

Perhaps I should reinstall the rom fresh because sadly, it keeps happening to me like every 20 minutes or so when trying to use the phone and switch between apps.

Perhaps you could advice what should I select in aroma. Like I mentioned, I didn't select Bixby, no dual speaker, I just did Bixby button as a Screenshot, some csc features and thats it. Really not sure, why is this happening for me.

**Comment by ananjaser1211** [comment]
> > no freezes on legit poweramp here, i also have cracked plex.
> > cant recreate any of the freezes and crashes yet
> 
> Perhaps I should reinstall the rom fresh because sadly, it keeps happening to me like every 20 minutes or so when trying to use the phone and switch between apps.
> 
> Perhaps you could advice what should I select in aroma. Like I mentioned, I didn't select Bixby, no dual speaker, I just did Bixby button as a Screenshot, some csc features and thats it. Really not sure, why is this happening for me.

Yeah i have no idea yet. i am on note 8 , i pick bixby vision, voice, home, gamehome/tools and chrome.

since you are on an S8 You cant do that. maybe try to stick with chrome for bloatware only.

but again, this really does not explain the crashes i saw in your log.

**Comment by nurikan** [comment]
> Since it is periodic, could you use logging kit to keep the phone recording, so when it crashes it gets captured.
> 
> in OPs log, his phone is trying to load an icon on the launcher, that is not really an icon. and it crashes TW Home & the systemUI
> 
> ```
> 12-29 23:46:56.972 29031 30363 E IconLoader: Could not find icon drawable from resource
> 
> 12-29 23:46:56.972 29031 30363 E IconLoader: android.content.res.Resources$NotFoundException: Resource ID #0xffffffff
> 
> 12-29 23:46:56.972 29031 30363 E IconLoader: 	at android.content.res.ResourcesImpl.getValueForDensity(ResourcesImpl.java:285)
> ```
> 
> Since none of these apps are modified in any way, and this error only seems to happen to few (3 so far, 1 on telegram who had themes), i have to assume it is some app that gets installed, or theme that causes this.
> 
> so we will need alot more info, including an active log to capture the crash

``

> Since it is periodic, could you use logging kit to keep the phone recording, so when it crashes it gets captured.
> 
> in OPs log, his phone is trying to load an icon on the launcher, that is not really an icon. and it crashes TW Home & the systemUI
> 
> ```
> 12-29 23:46:56.972 29031 30363 E IconLoader: Could not find icon drawable from resource
> 
> 12-29 23:46:56.972 29031 30363 E IconLoader: android.content.res.Resources$NotFoundException: Resource ID #0xffffffff
> 
> 12-29 23:46:56.972 29031 30363 E IconLoader: 	at android.content.res.ResourcesImpl.getValueForDensity(ResourcesImpl.java:285)
> ```
> 
> Since none of these apps are modified in any way, and this error only seems to happen to few (3 so far, 1 on telegram who had themes), i have to assume it is some app that gets installed, or theme that causes this.
> 
> so we will need alot more info, including an active log to capture the crash

#
[Log_2020-12-30_16-44-21.txt](https://github.com/ananjaser1211/hadesRom_Reports/files/5754883/Log_2020-12-30_16-44-21.txt)
Look somewhere at 20:02 please 
Idk if i could catch it

**Comment by Lincoln44** [comment]
I am going to reinstall the rom fresh again without root like I did before. Will report if something has changed in a few days or so, so don't assume I left this thread guys.

**Comment by nurikan** [comment]
> I am going to reinstall the rom fresh again without root like I did before. Will report if something has changed in a few days or so, so don't assume I left this thread guys.

I have same issue(freeze) without root, bloatware(only bixby but it not installed after flash), double speaker.

**Comment by nurikan** [comment]
> > Since it is periodic, could you use logging kit to keep the phone recording, so when it crashes it gets captured.
> > in OPs log, his phone is trying to load an icon on the launcher, that is not really an icon. and it crashes TW Home & the systemUI
> > ```
> > 12-29 23:46:56.972 29031 30363 E IconLoader: Could not find icon drawable from resource
> > 
> > 12-29 23:46:56.972 29031 30363 E IconLoader: android.content.res.Resources$NotFoundException: Resource ID #0xffffffff
> > 
> > 12-29 23:46:56.972 29031 30363 E IconLoader: 	at android.content.res.ResourcesImpl.getValueForDensity(ResourcesImpl.java:285)
> > ```
> > 
> > 
> > Since none of these apps are modified in any way, and this error only seems to happen to few (3 so far, 1 on telegram who had themes), i have to assume it is some app that gets installed, or theme that causes this.
> > so we will need alot more info, including an active log to capture the crash
> 
> ``
> 
> > Since it is periodic, could you use logging kit to keep the phone recording, so when it crashes it gets captured.
> > in OPs log, his phone is trying to load an icon on the launcher, that is not really an icon. and it crashes TW Home & the systemUI
> > ```
> > 12-29 23:46:56.972 29031 30363 E IconLoader: Could not find icon drawable from resource
> > 
> > 12-29 23:46:56.972 29031 30363 E IconLoader: android.content.res.Resources$NotFoundException: Resource ID #0xffffffff
> > 
> > 12-29 23:46:56.972 29031 30363 E IconLoader: 	at android.content.res.ResourcesImpl.getValueForDensity(ResourcesImpl.java:285)
> > ```
> > 
> > 
> > Since none of these apps are modified in any way, and this error only seems to happen to few (3 so far, 1 on telegram who had themes), i have to assume it is some app that gets installed, or theme that causes this.
> > so we will need alot more info, including an active log to capture the crash
> 
> [Log_2020-12-30_16-44-21.txt](https://github.com/ananjaser1211/hadesRom_Reports/files/5754883/Log_2020-12-30_16-44-21.txt)
> Look somewhere at 20:02 please
> Idk if i could catch it

[Log_2020-12-31_05-26-02.txt](https://github.com/ananjaser1211/hadesRom_Reports/files/5756256/Log_2020-12-31_05-26-02.txt)
Freeze again and new log
Look maybe you will find something same in these two (another sent in quoted message)

**Comment by Lincoln44** [comment]
@corsicanu @ananjaser1211 

Ok guys here is the thing:

I reinstalled the rom around 9 hours ago. Till then, I didn't have any freez at all. I didn't select any mods, no root, etc. I will upload the pictures of what I selected in aroma during the instalation process. This time, I installed Bixby tho just because I wanted, not sure what was the real reason haha. 

Anyway, now my thinking which I think is important. Devs, don't laught, it just my thinking and perhaps that would help.

Now I dont have any good lock modules installed. What I think, again( this is my thinking) when I installed the rom first time this is what I selected and this is what I had installed.

I had one hand operation installed and used it because I liked swipe up, back and down features, which I selected however I wanted. 

Next, I had selected LTE only network mode and force rcs support in aroma. Now, when I reinstalled the rom I didn't select those at all. See the pictures of what I selected during 2nd instalation.

So to sump up, I have been using the rom of 2nd instalation for 9 hours and I dont have anything at all of what I had experienced before. So my thinking is this:

It's either those 2 features that caused that lag, or good lock one handed operations, or both. Not sure.

If its silly thinking, I am sorry. Just wondering, what would devs say now after they see my message.

So far, everything is good and smooth. Again, no root, no mods, nothing.![Screenshot_20201231-022240_WhatsApp.jpg](https://user-images.githubusercontent.com/76757296/103391292-a84b5000-4b10-11eb-9ca4-8ec206afe948.jpg)![Screenshot_20201231-022235_WhatsApp.jpg](https://user-images.githubusercontent.com/76757296/103391294-abded700-4b10-11eb-8705-ca9161cbdc6e.jpg)![Screenshot_20201231-022230_WhatsApp.jpg](https://user-images.githubusercontent.com/76757296/103391297-ae413100-4b10-11eb-984e-00cb21b7b36a.jpg)![Screenshot_20201231-022224_WhatsApp.jpg](https://user-images.githubusercontent.com/76757296/103391300-b00af480-4b10-11eb-9288-e5e15a1a93dd.jpg)![Screenshot_20201231-022218_WhatsApp.jpg](https://user-images.githubusercontent.com/76757296/103391301-b13c2180-4b10-11eb-80f7-97eabbbb500a.jpg)![Screenshot_20201231-022214_WhatsApp.jpg](https://user-images.githubusercontent.com/76757296/103391302-b305e500-4b10-11eb-8917-905ba03351a3.jpg)

**Comment by ananjaser1211** [comment]
> @corsicanu @ananjaser1211
> 
> Ok guys here is the thing:
> 
> I reinstalled the rom around 9 hours ago. Till then, I didn't have any freez at all. I didn't select any mods, no root, etc. I will upload the pictures of what I selected in aroma during the instalation process. This time, I installed Bixby tho just because I wanted, not sure what was the real reason haha.
> 
> Anyway, now my thinking which I think is important. Devs, don't laught, it just my thinking and perhaps that would help.
> 
> Now I dont have any good lock modules installed. What I think, again( this is my thinking) when I installed the rom first time this is what I selected and this is what I had installed.
> 
> I had one hand operation installed and used it because I liked swipe up, back and down features, which I selected however I wanted.
> 
> Next, I had selected LTE only network mode and force rcs support in aroma. Now, when I reinstalled the rom I didn't select those at all. See the pictures of what I selected during 2nd instalation.
> 
> So to sump up, I have been using the rom of 2nd instalation for 9 hours and I dont have anything at all of what I had experienced before. So my thinking is this:
> 
> It's either those 2 features that caused that lag, or good lock one handed operations, or both. Not sure.
> 
> If its silly thinking, I am sorry. Just wondering, what would devs say now after they see my message.
> 
> So far, everything is good and smooth. Again, no root, no mods, nothing.!

i will look into the force RCS thing, its the only thing i dont have enabled. i use hand+ module , sound assistant and edge+ modules on my Note 8.

i highly doubt they are causing the crash, but who knows.

i am now wondering if its some user app, or some old data in `Android/data` on your sdcard that might be responsible. just a thought.

one more thing is, what is the common thing between you and @SoRadGaming , because so far you are the two with these issues. but i cant figure out what is common. if its certain apps, installation method. i dont know honestly. 

maybe if we can find out how you two have the same crash, we might be able to isolate the cause.

**Comment by Lincoln44** [comment]
It's fine by me. I will help in any way I can. I will try installing some of goodlock modules because I like a few of them and will report the behavior.

@ananjaser1211 

Ok, I have been using one hand operation for 5 hours or so, and I haven't experienced lags of freezes at all like before. Thus, I dont think that was the issue.

So, my guess is it was LTE only network mode and or force rcs support. I am using my phone without any mods, without any tinkering, etc. So, I really don't know what could have caused that even during the first instalation. Really weird.

**Comment by Snowers96** [comment]
Hi! If I select force RCS and uncheck LTE icon, the ROM lags terribly.
But, uncheck force RCS and checking LTE icon fix the problem

**Comment by nurikan** [comment]
> Hi! If I select force RCS and uncheck LTE icon, the ROM lags terribly.
> But, uncheck force RCS and checking LTE icon fix the problem

Are there any way is doing this two without flash?

**Comment by Snowers96** [comment]
> > Hi! If I select force RCS and uncheck LTE icon, the ROM lags terribly.
> > But, uncheck force RCS and checking LTE icon fix the problem
> 
> Are there any way is doing this two without flash?

I don't know, but for testing this workaround better if you flash again the ROM

**Comment by SoRadGaming** [comment]
My phone freezes are only 15 seconds long, everything is unresponsive, its really annoying because I cant consistently recreate it. One time I open Snapchat and record, zoom and the whole phone just rebooted. I did check force RCS, and later learned my carrier promised RCS rollout in 2017 but has not done it yet (Optus)

**Comment by ananjaser1211** [comment]
did anyone try @Snowers96 suggestion. if it works we can handle that through aroma

**Comment by corsicanu** [comment]
@ananjaser1211 I use forced RCS and no LTE icon and no lags/freezes so they shouldn't be related to eachother.

**Comment by Snowers96** [comment]
> @ananjaser1211 I use forced RCS and no LTE icon and no lags/freezes so they shouldn't be related to eachother.

Could be that for example, some providers don't support RCS yet? Because I know another guy from Argentina that have this problem, and we live in a country where RCS is not enabled

**Comment by Lincoln44** [comment]
@corsicanu @ananjaser1211 

I did a bit of testing in the past few days. Here is what I found: 

I installed one hand operation and was using it. Then, I uninstalled it and was using my phone again. Both times ( with hands operations and without) i experienced some lags and freezes. Nothing major, like before described in op. When trying to open some apps or multitask, phone hangs for 5 to 10 seconds and thats it. Quick point. I was using goodlock from galaxy store, not the patched one from aroma if that makes any difference.

Also, when going to media or devices in notifications and switching from Bluetooth sound to device or vice versa and then swiping home, nothing appears. My homescreen is empty. When swiping again home, icons and widgets appear. Just wanted to point this out.

So what now. Should I root my phone and try to capture those lags again and post the logs ?

P.s. This issue a few of us are having is really weird. Hope, we haven't pissed u off guys 😀

**Comment by SypeR54** [comment]
@corsicanu In some countries RCS not enabled on all carriers. If carrier doesn't support RCS, probably forcing it causes spamming the modem in the background. And spam == filled up cache. Maybe thats the reason of lags.

**Comment by SoRadGaming** [comment]
> I installed one hand operation and was using it. Then, I uninstalled it and was using my phone again. Both times ( with hands operations and without) i experienced some lags and freezes. Nothing major, like before described in op. When trying to open some apps or multitask, phone hangs for 5 to 10 seconds and thats it. Quick point. I was using goodlock from galaxy store, not the patched one from aroma if that makes any difference.

I did not use or install goodlock or onehand and I still get phone hangs for 5 to 10 seconds, it happens most often when apps use the camera2api. I want to find a consistent way to recreate it first, then ill try the RCS off fix.

> Also, when going to media or devices in notifications and switching from Bluetooth sound to device or vice versa and then swiping home, nothing appears. My homescreen is empty. When swiping again home, icons and widgets appear. Just wanted to point this out.

I don't have this Issue.

**Comment by SoRadGaming** [comment]
Ok reflashed the rom with force RCS off, and am still getting the lag for 10-15 seconds, tho I have figured a way to recreate it. Simply open an app wait for it to load then close and open another app, keep doing this and I assume at some point the phones ram management fucks up and the phone freezes for a solid 5-10 seconds.

**Comment by SoRadGaming** [comment]
Another way to create a system freeze was opening Youtube (Vanced) loading a video then rotate the phone to landscape, the phone sometimes just freezes and nothing works, not event he power button, and you have to wait 7ish seconds then it responds for 1-second freezes again then after 6 seconds everything back to normal.

Edit: Another Method is to open Instagram, swipe to camera then close the app quickly, the system freezes for 7-15 seconds. What should I use to get a good log reading, science I can recreate the issue?

**Comment by ananjaser1211** [comment]
i have tried all the mentioned methods above and i still cant recreate a single freeze / lag.. 

the only log that would help is log_all.txt in loggingkit, since you can recreate it, i prefer you do the instagram method, as it is the least to generate alot of spam.

run logcat_all.bat file while your phone is connected, follow loggingkit instructions if you got stuck, and upload the log here.

**Comment by SoRadGaming** [comment]
[android-debug.log](https://github.com/ananjaser1211/hadesRom_Reports/files/5763388/android-debug.log)
I ran adb logcat all deviceID > C:\android-debug.log, in platform-tools, hoping that what I'm supposed to do, as soon as it started running opened Instagram and the phone freezer for around 6 seconds then once it was good stoped the log with ctrl+c.

**Comment by SoRadGaming** [comment]
Another one
[Crash-Instagram.log](https://github.com/ananjaser1211/hadesRom_Reports/files/5763400/Crash-Instagram.log)

**Comment by ananjaser1211** [comment]
@SoRadGaming going through the logs, here is what i see.

1) on boot, bixby button remapper `com.jamworks.bxactions` is crashing, so try to remove it for the next tests

2) The lag you face seems to happen when camera is turned on, based on the two logs, but the cause is not yet clear

what i see is, after camera initializes, the phone runs out of ram and Lowmemorykiller reclaims some ram by killing `com.samsung.android.dynamiclock`


This snippet is while instagram is running

```
01-04 17:49:35.757  4514  4772 I libprocessgroup: Successfully killed process cgroup uid 10137 pid 3271 in 0ms
01-04 17:49:35.780  4092  4092 I lmkd    : 17(delay),0(swap), 0(freelimit) memory pressure events were skipped after a kill!
01-04 17:49:35.781  4092  4092 I lmkd    : cached 0, sandbox(not0) 0
01-04 17:49:35.781  4056  4056 I Zygote  : Process 3271 exited due to signal 9 (Killed)
01-04 17:49:35.784  4092  4092 I lmkd    : Reclaim Mem 'com.samsung.android.dynamiclock' (1334), uid 10075, oom_adj 925, state 20 to free 18524kB
01-04 17:49:35.784  4092  4092 I lmkd    : Reclaimed 18524kB at oom_adj 850
```

in fact, there is ALOT of lmkd kills..

can you please remind me again, if you are using a note 8 or an S8* . i would like if you can intall hktweaks, open virtualmemory tab, and before you open instagram take a screenshot, then after the lag happens take another.

i have a feeling your swap / memory will be almost if not full. assuming LMK does not kill the apps before we get a chance to see the real usage.

might help to send device maintenance screenshot of ram page, what recently used it etc

**Comment by SoRadGaming** [comment]
Uninstalled BxActions, I do still get lag from exiting non-camera apps(Reddit/Youtube Vanced), tho I can't recreate them all the time. I'm on S8+(sm-g955f).
![Screenshot_20210105-142332_Device care](https://user-images.githubusercontent.com/47128515/103603274-5700ed80-4f62-11eb-8348-49e882e9accb.jpg)
![Screenshot_20210105-142340_Device care](https://user-images.githubusercontent.com/47128515/103603277-58321a80-4f62-11eb-93bf-1cefae37c00a.jpg)
![Screenshot_20210105-142347_Device care](https://user-images.githubusercontent.com/47128515/103603278-58cab100-4f62-11eb-9431-76f90dc32dc1.jpg)
![Screenshot_20210105-142350_Device care](https://user-images.githubusercontent.com/47128515/103603280-59634780-4f62-11eb-9f26-683e0600dcbd.jpg)
![Screenshot_20210105-142358_Device care](https://user-images.githubusercontent.com/47128515/103603281-59fbde00-4f62-11eb-812c-10dc4d541749.jpg)
![Screenshot_20210105-142418_Settings](https://user-images.githubusercontent.com/47128515/103603284-5a947480-4f62-11eb-97f9-9e6272ebe0f7.jpg)
![Screenshot_20210105-142424_Settings](https://user-images.githubusercontent.com/47128515/103603285-5a947480-4f62-11eb-9ba0-9fd9c6809225.jpg)
![Screenshot_20210105-142431_Settings](https://user-images.githubusercontent.com/47128515/103603286-5b2d0b00-4f62-11eb-98fb-5bfb00d30c1c.jpg)
![Screenshot_20210105-142440_Settings](https://user-images.githubusercontent.com/47128515/103603288-5bc5a180-4f62-11eb-8251-c1d549b4d4df.jpg)
![Screenshot_20210105-142445_Settings](https://user-images.githubusercontent.com/47128515/103603290-5c5e3800-4f62-11eb-902c-2bdf4d82951b.jpg)
Before
![Screenshot_20210105-142605_hKtweaks](https://user-images.githubusercontent.com/47128515/103603291-5c5e3800-4f62-11eb-934d-28bb5a131770.jpg)
After
![Screenshot_20210105-142634_hKtweaks](https://user-images.githubusercontent.com/47128515/103603292-5cf6ce80-4f62-11eb-8e4c-2b9f72271527.jpg)

**Comment by ananjaser1211** [comment]
ok so this is a start. its a ram issue of sorts, but here is how the behavior on S8 differs from the note 8 (and why we cant recreate this issue)

in S8/S8+ , due to 4GB ram, the phone is prioritizing swap memory most of the time it seems, when the camera runs, notice how instead of filling RAM, it filled the swap memory, 

Swap memory (zram in this case) is compressed too, which will use cpu % , so this might explain why the phone would freeze, as it is compressing.

in any case, swap should never be this full, and RAM should be prioritized more often.

so i have few tests to ask you to do, that MIGHT resolve this issue

**### First method**, reboot the phone (so swap is "empty") , open hktweaks app, turn off ZRAM in virtual memory.

HKtweaks and partially your phone will freeze as ZRAM is being purged into RAM. HKTweaks might show "FC" few times if you try to interact with the frozen app, press "continue waiting"

Id say give it 2 or 3 minutes, then go back to home, and close the app / reopen it

if you go to virtualmemory you will see that it will say 100% used, but the size is decreasing, that is basically zram purging.

Now ,  you either wait for it to fully purge zram, or try instagram again, it might "lag" a bit due to decompression, but in theory, it should no longer open the camera in zram causing the freeze and lags you had.

in theory.

**### Second method** , this is what i personally recommend, but i do not know if its going to help remedy this issue 100%.

go to hktweaks > virtual memory and change swappiness value from 160 to 100.
test your instagram now. if its the same, change it to 50.

it should bias RAM now instead of zram.

please test both methods and let me know your findings, if you can share hktweaks virtualmemory screenshots with zram disabled. with before/after opening instragram, that would be great too.

I know it is long, but it will help alot. thanks

**Comment by SoRadGaming** [comment]
Set virtual memory swappiness value from 160 to 100, same lag after a system restart (No lag change).
Before
![Screenshot_20210105-162300_hKtweaks](https://user-images.githubusercontent.com/47128515/103609671-77d13f00-4f72-11eb-91fa-264b1a90a2a9.jpg)
After
![Screenshot_20210105-162327_hKtweaks](https://user-images.githubusercontent.com/47128515/103609683-7f90e380-4f72-11eb-997d-955db0d20ef4.jpg)
Trying Method 1, turn Zram off, phone froze and just kept clicking wait (swappiness value still at 100), once it was good again took this screenshot of it emptying
![Screenshot_20210105-163209_hKtweaks](https://user-images.githubusercontent.com/47128515/103610245-c6cba400-4f73-11eb-8e86-117d40adfa32.jpg)
Before Instagram Freeze
![Screenshot_20210105-164346_hKtweaks](https://user-images.githubusercontent.com/47128515/103611035-84a36200-4f75-11eb-80f5-0ea29710c77a.jpg)
After
![Screenshot_20210105-164348_hKtweaks](https://user-images.githubusercontent.com/47128515/103611054-8ff68d80-4f75-11eb-8638-de9aca85cf34.jpg)
Tried Method 1 and 2 at the same time with 50 memory swappiness value, the lag is much less, still freezes for around 3-6 seconds compared to 12 seconds
Before
![Screenshot_20210105-170546_hKtweaks](https://user-images.githubusercontent.com/47128515/103612449-90dcee80-4f78-11eb-8bd3-3c2ab02956bc.jpg)
After
![Screenshot_20210105-170619_hKtweaks](https://user-images.githubusercontent.com/47128515/103612459-95a1a280-4f78-11eb-8a25-8adc8142200b.jpg)

**Comment by SoRadGaming** [comment]
Tried Method 1 and 2 at the same time with 50 memory swappiness value again, this time leaving the phone for a bit after the restart. The lag was even worse, the phone ended up rebooting after like 4 minutes of the system being frozen. Was only able to get a before image.
![Screenshot_20210105-172437_hKtweaks](https://user-images.githubusercontent.com/47128515/103614104-f5e61380-4f7b-11eb-96d0-e3455551d5c2.jpg)
The Phone just keeps freezing now, going to reset it.
Edit: Using only 50 memory swappiness modification has reduced lag time, and been most effective. The issue still persists for 3 seconds now instead of the 12-second lag.

**Comment by ananjaser1211** [comment]
its due to apply on boot, reboot your phone, once it boots drag the qs toggle and press cancel on the applying settings notif by hktweaks.
ill see the previous screenshots in a while

**Comment by ananjaser1211** [comment]
This log over here contains what actually crashes and takes down SystemUI and other services with it. it is some drawable bug like here https://github.com/ananjaser1211/hadesRom_Reports/issues/17#issuecomment-752415575 

not really clear what causes it on S8/S8+ only (and not N8). but atleast we know where to look now

[Skia.log](https://github.com/ananjaser1211/hadesRom_Reports/files/5824286/Skia.log)

**Comment by Lincoln44** [comment]
Awesome news @ananjaser1211 . Hopefully it could be reduced to a minimum if not fully fixed in the next release.

P.s. after my 2nd instalation, it got better. For a few days I did not get any lags or freezes. I do get them occasionally tho. But other than that, the rom is 👌.

**Comment by mehmetgunacti** [comment]
Since this is labelled as "addressed" ("fixed for next release") is it possible to provide a hot-fix for the freeze problem until v1.1 comes out?
